cameraview/src/main/java/com/otaliastudios/cameraview/engine/Camera2Engine.java

```diff
@@ -1252,6 +1256,45 @@ public void setPlaySounds(boolean playSounds) {
         mPlaySoundsOp.end(null);
     }
 
+    @Override public void setPreviewFrameRate(float previewFrameRate) {
+        final float oldPreviewFrameRate = mPreviewFrameRate;
+        mPreviewFrameRate = previewFrameRate;
+        mHandler.run(new Runnable() {
+            @Override
+            public void run() {
+                if (getEngineState() == STATE_STARTED) {
+                    if (applyPreviewFrameRate(mRepeatingRequestBuilder, oldPreviewFrameRate)) {
+                        applyRepeatingRequestBuilder();
+                    }
+                }
+                mPreviewFrameRateOp.end(null);
+            }
+        });
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    protected boolean applyPreviewFrameRate(@NonNull CaptureRequest.Builder builder, float oldPreviewFrameRate) {
+        Range<Integer> range = new Range<>((int) mCameraOptions.getFpsRangeMinValue(), (int) mCameraOptions.getFpsRangeMaxValue());
+        if (mPreviewFrameRate != 0f) {
+            if (range.contains((int) mPreviewFrameRate)) {
+                builder.set(CaptureRequest.CONTROL_AE_TARGET_FPS_RANGE, range);
+                return true;
+            }
```

Comment:
I still think that this can crash. You should get the ranges from `CONTROL_AE_AVAILABLE_TARGET_FPS_RANGES` and get the first one that contains `mPreviewFrameRate` or we can create an invalid range because min and max value of `CameraOptions` might come from different `Range` objects

Reply:
@natario1 that's what I had earlier I was using set of ranges in CameraOptions and which contained the mPreviewFrameRate I was using that rather than using min and max in CameraOptions

Reply:
@natario1 that's what I had earlier I was using set of ranges in CameraOptions and which contained the mPreviewFrameRate I was using that rather than using min and max in CameraOptions